src/Components/EventDetail.jsx

```diff
@@ -0,0 +1,157 @@
+    if (auth.authorization) {
+      fetchWithAuth();
+    }
+  }, [id, dispatch]);
+
+  if (auth.authorization === false) {
+    return (
+      <Typography.Text>
+        You cannot access this page without log in, please click log in first.
+      </Typography.Text>
+    );
+  }
+  if (eventDetail === null) {
+    fetchWithAuth();
+    return <div>Cannot fetch event detail</div>;
+  }
+
+  const rate =
+    eventDetail.reviews === null
+      ? "NaN"
+      : eventDetail.reviews.reduce(sumReduce, 0) / eventDetail.reviews.length;
+
+  return (
+    <Row>
+      <Col key={nanoid()} span={14}>
+        <Card
+          key={nanoid()}
+          title={<div className="eventcard-title">{eventDetail.title}</div>}
+          className="eventcard-body"
+        >
+          <Row align="space-between" className="eventcard-row">
+            <Col>Category: {eventDetail.category}</Col>
+            <Col>
+              <IconText icon={StarOutlined} text={rate}></IconText>
+            </Col>
+          </Row>
+          <Card.Grid
+            key={nanoid()}
+            className="eventcard-centergrid"
+            hoverable={false}
+          >
+            <Image src={eventDetail.picture_url} alt="Event_image" />
+          </Card.Grid>
+          <Card.Grid
+            key={nanoid()}
+            className="eventcard-leftgrid"
+            hoverable={false}
+          >
+            Description: {eventDetail.description}
+          </Card.Grid>
+          <Card.Grid
+            key={nanoid()}
+            className="eventcard-leftgrid"
+            hoverable={false}
+          >
+            <Avatar src={eventDetail.host.profile_picture_url} />
+            <span className="eventcard-subtitle">
+              &nbsp;&nbsp; Hostname: {eventDetail.host.first_name}{" "}
+              {eventDetail.host.last_name}
+            </span>
+          </Card.Grid>
+          <Card.Grid
+            key={nanoid()}
+            className="eventcard-leftgrid"
+            hoverable={false}
+          >
+            <span className="eventcard-subtitle">Who are coming:</span>
+            <br></br>
+            <Space size={20}>
+              {eventDetail.attendees === null
+                ? "No body will come"
+                : eventDetail.attendees.map((ele) => (
+                    <Tooltip
+                      title={`id: ${ele.id}`}
+                      placement="top"
+                      key={ele.id}
+                    >
+                      <Avatar key={nanoid()} src={ele.profile_picture_url} />
+                    </Tooltip>
+                  ))}{" "}
+            </Space>
+          </Card.Grid>
+          <Card.Grid className="eventcard-leftgrid" hoverable={false}>
+            <h4>What do other people think about this event?</h4>
+          </Card.Grid>
+          {eventDetail.reviews?.map((review) => (
+            <Card.Grid className="eventcard-leftgrid" hoverable={false}>
+              <Row align="space-between" key={nanoid()}>
```

Comment:
It's better to use the review id rather than generating a new one. The reason is if you generate a new one on every render, React will treat it as a new element, then React will render the whole list again. So it's defeating the purpose of DOM rendering optimization.